### PR TITLE
[Bug]: Visible fields on many-to-many object relations are not shown.

### LINKED ISSUE #13439: [Bug]: Visible fields on many-to-many object relations are not shown.
### Expected behavior

When enabling visible fields on a many-to-many object relation. The visible fields should be shown on the object:
image

### Actual behavior

**Only two of the selected visible fields are shown properly:**

image

### Steps to reproduce

1. Add many-to-many object relation
2. Select multiple visible fields ( incl. image & boolean select fields)
3. Assign objects on the relation & check the visible fields.

### PR DESCRIPTION
**Pull request by mcop1**
Resolves #13439

### CODE REVIEW

**Review by kjkooistra-youwe — APPROVED** [review]

File: models/DataObject/ClassDefinition/Data/ManyToManyObjectRelation.php [hunk 1]
```diff
@@ -636,6 +637,10 @@ public function enrichLayoutDefinition(/* ?Concrete */ $object, /* array */ $con
 
                     $this->visibleFieldDefinitions[$field]['options'] = $fd->getOptions();
                 }
+
+                if ($fd instanceof DataObject\ClassDefinition\Data\BooleanSelect){
+                    $this->visibleFieldDefinitions[$field]['options'] = $fd->getOptions();
+                }
```
Comment (mcop1) [inline models/DataObject/ClassDefinition/Data/ManyToManyObjectRelation.php]: Do we have to add those lines for other relations too? E.g. here https://github.com/pimcore/pimcore/blob/10.5/models/DataObject/ClassDefinition/Data/AdvancedManyToManyObjectRelation.php#L862-L871 
Possibly in other places as well?

File: bundles/AdminBundle/Resources/public/js/pimcore/object/tags/advancedManyToManyObjectRelation.js [hunk 2]
```diff
@@ -133,7 +133,9 @@ pimcore.object.tags.advancedManyToManyObjectRelation = Class.create(pimcore.obje
 
                 if (fc.layout.key === "fullpath") {
                     fc.renderer = this.fullPathRenderCheck.bind(this);
-                } else if(fc.layout.layout.fieldtype == "select" || fc.layout.layout.fieldtype == "multiselect") {
+                } else if(fc.layout.layout.fieldtype == "select" ||
```
Comment (kjkooistra-youwe) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/tags/advancedManyToManyObjectRelation.js]: Preferably add a space behind the if and add the operators before the condition. Optionally use single quotes for strings (consistent with other comment).
```javascript
} else if (
    fc.layout.layout.fieldtype === 'select'
    || fc.layout.layout.fieldtype === 'multiselect'
    || fc.layout.layout.fieldtype === 'booleanSelect'
) {
```

File: bundles/AdminBundle/Resources/public/js/pimcore/object/tags/manyToManyObjectRelation.js [hunk 3]
```diff
@@ -344,7 +344,10 @@ pimcore.object.tags.manyToManyObjectRelation = Class.create(pimcore.object.tags.
 
                 if(fc.layout.key === "fullpath") {
                     fc.renderer = this.fullPathRenderCheck.bind(this);
-                } else if(fc.layout.layout.fieldtype == "select" || fc.layout.layout.fieldtype == "multiselect") {
+                }
```
Comment (kjkooistra-youwe) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/tags/manyToManyObjectRelation.js]: Curly brace should be before the else (consistent with other JavaScript).

File: bundles/AdminBundle/Resources/public/js/pimcore/object/tags/manyToManyObjectRelation.js [hunk 4]
```diff
@@ -344,7 +344,10 @@ pimcore.object.tags.manyToManyObjectRelation = Class.create(pimcore.object.tags.
...
+                else if(fc.layout.layout.fieldtype == "select" ||
```
Comment (kjkooistra-youwe) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/tags/manyToManyObjectRelation.js]: Preferably use format and consistent (single?) quotes for strings:
```php
} else if (
    fc.layout.layout.fieldtype === 'select'
    || fc.layout.layout.fieldtype === 'multiselect'
    || fc.layout.layout.fieldtype === 'booleanSelect'
) {
```

File: models/DataObject/ClassDefinition/Data/AdvancedManyToManyObjectRelation.php [hunk 5]
```diff
@@ -857,14 +857,17 @@ public function enrichLayoutDefinition(/* ?Concrete */ $object, /* array */ $con
                     $this->visibleFieldDefinitions[$field]['title'] = $translator->trans($field, [], 'admin');
                     $this->visibleFieldDefinitions[$field]['fieldtype'] = 'input';
                 }
-            } else {
+            }
```
Comment (kjkooistra-youwe) [inline models/DataObject/ClassDefinition/Data/AdvancedManyToManyObjectRelation.php]: Curly brace should be before the else.

File: models/DataObject/ClassDefinition/Data/AdvancedManyToManyObjectRelation.php [hunk 6]
```diff
@@ -857,14 +857,17 @@ public function enrichLayoutDefinition(/* ?Concrete */ $object, /* array */ $con
...
+            else {
                 $this->visibleFieldDefinitions[$field]['name'] = $fd->getName();
                 $this->visibleFieldDefinitions[$field]['title'] = $fd->getTitle();
                 $this->visibleFieldDefinitions[$field]['fieldtype'] = $fd->getFieldType();
                 $this->visibleFieldDefinitions[$field]['noteditable'] = true;
 
-                if ($fd instanceof DataObject\ClassDefinition\Data\Select || $fd instanceof DataObject\ClassDefinition\Data\Multiselect) {
-                    if ($fd->getOptionsProviderClass()) {
+                if ($fd instanceof DataObject\ClassDefinition\Data\Select ||
```
Comment (kjkooistra-youwe) [inline models/DataObject/ClassDefinition/Data/AdvancedManyToManyObjectRelation.php]: Preferably format similar to (for readability):
```php
if (
    $fd instanceof DataObject\ClassDefinition\Data\Select
    || $fd instanceof DataObject\ClassDefinition\Data\Multiselect
    || $fd instanceof DataObject\ClassDefinition\Data\BooleanSelect
) {
```

File: models/DataObject/ClassDefinition/Data/ManyToManyObjectRelation.php [hunk 7]
```diff
@@ -630,17 +630,15 @@ public function enrichLayoutDefinition(/* ?Concrete */ $object, /* array */ $con
                 $this->visibleFieldDefinitions[$field]['fieldtype'] = $fd->getFieldType();
                 $this->visibleFieldDefinitions[$field]['noteditable'] = true;
 
-                if ($fd instanceof DataObject\ClassDefinition\Data\Select || $fd instanceof DataObject\ClassDefinition\Data\Multiselect) {
-                    if ($fd->getOptionsProviderClass()) {
+                if ($fd instanceof DataObject\ClassDefinition\Data\Select ||
```
Comment (kjkooistra-youwe) [inline models/DataObject/ClassDefinition/Data/ManyToManyObjectRelation.php]: Same formatting as other comment.

**Comment by Corepex** [comment]
@kjkooistra-youwe thanks for the very detailed feedback - i have reformatted the code and implemented some of your suggestions.

File: bundles/AdminBundle/Resources/public/js/pimcore/object/tags/manyToManyObjectRelation.js [hunk 8]
```diff
@@ -342,9 +342,11 @@ pimcore.object.tags.manyToManyObjectRelation = Class.create(pimcore.object.tags.
                 fc.editor = null;
                 fc.sortable = false;
 
-                if(fc.layout.key === "fullpath") {
+                if (fc.layout.key === "fullpath") {
                     fc.renderer = this.fullPathRenderCheck.bind(this);
-                } else if(fc.layout.layout.fieldtype == "select" || fc.layout.layout.fieldtype == "multiselect") {
+                } else if (fc.layout.layout.fieldtype == 'select'
+                    || fc.layout.layout.fieldtype == 'multiselect'
+                    ||fc.layout.layout.fieldtype == 'booleanSelect') {
```
Comment (kjkooistra-youwe) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/tags/manyToManyObjectRelation.js]: There should be a space between `||` and fc

Adding `) {` to the next line will separate the condition from the code below more clearly.

File: bundles/AdminBundle/Resources/public/js/pimcore/object/tags/booleanSelect.js [hunk 9]
```diff
@@ -34,9 +34,11 @@ pimcore.object.tags.booleanSelect = Class.create(pimcore.object.tags.abstract, {
                 }
             }
 
-            for(var i=0; i<field.layout.options.length; i++) {
-                if(field.layout.options[i]["value"] == value) {
-                    return field.layout.options[i]["key"];
+            if(field.layout.options !== undefined) {
```
Comment (kjkooistra-youwe) [inline bundles/AdminBundle/Resources/public/js/pimcore/object/tags/booleanSelect.js]: There should be a space between `if` and `(` (and preferably also between `for` and `(` below).

File: bundles/AdminBundle/Resources/public/js/pimcore/object/tags/manyToManyObjectRelation.js (hunk 8 above)
Comment (Corepex) [reply bundles/AdminBundle/Resources/public/js/pimcore/object/tags/manyToManyObjectRelation.js]: true i must have missed this one - thanks for commenting

File: models/DataObject/ClassDefinition/Data/AdvancedManyToManyObjectRelation.php (hunk 6 above)
Comment (kjkooistra-youwe) [reply models/DataObject/ClassDefinition/Data/AdvancedManyToManyObjectRelation.php]: As a sidenote: this is from the PSR-12 standard. I'm not sure if Pimcore is using this.
https://www.php-fig.org/psr/psr-12/#51-if-elseif-else

> Expressions in parentheses MAY be split across multiple lines, where each subsequent line is indented at least once. When doing so, the first condition MUST be on the next line. The closing parenthesis and opening brace MUST be placed together on their own line with one space between them. Boolean operators between conditions MUST always be at the beginning or at the end of the line, not a mix of both.

Comment (Corepex) [reply models/DataObject/ClassDefinition/Data/AdvancedManyToManyObjectRelation.php]: thanks for the hint - adjusted the code

Comment (kjkooistra-youwe) [reply models/DataObject/ClassDefinition/Data/AdvancedManyToManyObjectRelation.php]: @Corepex Thanks for the fixes.

**Pull request merged**